tad/services/storage.py

```diff
@@ -0,0 +1,32 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+
+from yaml import dump
+
+
+class WriteService(ABC):
+    def __init__(self, location: str, filename: str) -> None:
+        self.location = location
+        if not filename.endswith(".yaml"):
+            raise ValueError(f"Filename {filename} must end with .yaml instead of .{filename.split('.')[-1]}")
+        self.filename = filename
+
+    @abstractmethod
+    def write(self, data: dict) -> None:
+        pass
+
+
+class FileSystemWriteService(WriteService):
```

Comment:
Please add the close function implementation

Reply:
In the abstract class i implemented it but for the filesystemwriter as discussed it is not needed